File: app/src/main/java/android/montreal/devfest/stuff/presentation/activities/MainActivityModule.java
package android.montreal.devfest.stuff.presentation.activities;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

@Module(subcomponents = {MainActivitySubComponent.class})
public abstract class MainActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(MainActivity.class)
    abstract AndroidInjector.Factory<? extends Activity> injector(MainActivitySubComponent.Builder builder);
}

File: stuff_presentation/src/main/java/android/montreal/devfest/stuff/presentation/activities/MainActivityModule.kt
package android.montreal.devfest.stuff.presentation.activities

import android.app.Activity

import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap

@Module(subcomponents = [MainActivitySubComponent::class])
abstract class MainActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(MainActivity::class)
    internal abstract fun injector(builder: MainActivitySubComponent.Builder): AndroidInjector.Factory<out Activity>
}
